### PR TITLE
fix(runner): index wildcard source prefixes

### PR DESCRIPTION
**Pull request by seefeldb**
Wildcard sources currently send every query through a full scan in both CFC path indexes. Bucket those sources at the trie node before their first `*`, so concrete queries check only candidates sharing their prefix. Keep the wildcard-query scan and the shared `isPrefix` predicate, including overlap ordering, shorter queries, duplicate labels, and copied source paths.

The generated correctness grid covers 50/250/1000 sources, wildcard fractions 0/0.05/0.3, and query depths 2–7 for both indexes. Warm paired measurements put queries over wildcard sources within 1.68× of their concrete-source counterparts. At 1,000 sources, source-wildcard overlaps fall from 213–220 µs to 0.36–0.39 µs/query. The checked-in benchmark, raw measurements, and replay are in the [measurement report](docs/history/development/performance/2026-09-15-cfc-wildcard-source-index.md).

The shared arm-B ladder is reported with an explicit limitation: it does not establish an attributable end-to-end gain from these indexes. Over 98% of sampled `isPrefix` self time remains in the separate `authoritativeCoverFor` scan. This PR does not claim that bottleneck is fixed.

Validation: generated scan-equivalence tests and focused edge cases; full runner suite (1,427 tests / 10,099 steps); repo-wide formatting, lint, and type checks; documentation, history-index, conflict-marker, and control-character checks. Self-reviewed with `cf-review`; no blocking correctness or convention findings.

### CODE REVIEW

File: docs/history/development/performance/2026-09-15-cfc-wildcard-source-index.replay.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+}
+"""
+
+BASE = "53baf62fdaf9dae6824bc9c5a0da812a64b95d6c"
+checkout = Path(os.environ["CHECKOUT"]).resolve()
+with tempfile.TemporaryDirectory(prefix="cfc-wildcard-replay-") as directory:
+    scratch = Path(directory)
+    for name in ["path-prefix-index.ts", "consumed-label-index.ts"]:
+        source = subprocess.check_output(
+            ["git", "show", f"{BASE}:packages/runner/src/cfc/{name}"],
```
[the remark this answers is not in the document]
Comment (seefeldb) [reply docs/history/development/performance/2026-09-15-cfc-wildcard-source-index.replay.py:72]: Fixed in 126736b914. The replay documents its baseline-history requirement and validates the commit before reading any source files or starting measurements. If it is missing, the error prints an exact shell-quoted `git -C ... fetch origin <baseline>` command.

Verified with a depth-one clone: the replay exits early with that diagnostic, and the printed fetch restores access to the baseline commit. The full-history replay also completed successfully with all 126 measurement rows. — Codex
